### DISCUSSION TITLE
Syslog-ng Local Name Resolution

### DISCUSSION THREAD

**Discussion by garutialle** [syslog-ng/syslog-ng#4960] [General]
I'm using local name resolution, use-dns(persit_only) ok but the field that is modified is host_from and not host which remains the one sent by the device, is it not possible to modify host leaving host_from unchanged? Should I use a rewrite?

**Comment by MrAnno** [comment]
The `keep-hostname(yes/no)` source option is what you are probably looking for.
`$HOST` either keeps the original hostname parsed from the message, or is replaced by syslog-ng according to the keep-hostname, chain-hostname, etc. options.

`$HOST_FROM` always stores the sender's hostname (last hop).

**Reply by garutialle** [reply]
I already tested that option and it doesn't change anything. example I have a qnap hostname Archive this sends the logs with hostname Archive and even if disabled keep-hostname the host name is always Archive it modifies HOST_FROM

Thanks
